### PR TITLE
zigbee: Fix zb_osif_abort() implementation [KRKNWK-12354]

### PR DESCRIPTION
**Pull request by sebastiandraus**

### CODE REVIEW

File: doc/nrf/libraries/zigbee/osif.rst [hunk 1]
```diff
@@ -27,6 +27,10 @@ You can also configure the following OSIF-related Kconfig options:
   .. note::
       These functions are useful for debugging, but they can cause instability of the device.
 
+* :kconfig:`CONFIG_ZBOSS_RESET_ON_ASSERT` - Configures the ZBOSS OSIF layer to reset the device when ZBOSS assert occurs, enabled by default.
+  Use this option for production ready applications.
+* :kconfig:`CONFIG_ZBOSS_HALT_ON_ASSERT` - Configures the ZBOSS OSIF layer to halt the device when ZBOSS assert occurs.
+  Use this option only for testing and debugging your application.
```
Comment (greg-fer) [inline doc/nrf/libraries/zigbee/osif.rst:33]:
```suggestion
* :kconfig:`CONFIG_ZBOSS_RESET_ON_ASSERT` - Configures the ZBOSS OSIF layer to reset the device when a ZBOSS assert occurs.
  This option is enabled by default.
  Use it for production-ready applications.
* :kconfig:`CONFIG_ZBOSS_HALT_ON_ASSERT` - Configures the ZBOSS OSIF layer to halt the device when a ZBOSS assert occurs.
  Use this option only for testing and debugging your application.
```

File: subsys/zigbee/Kconfig [hunk 2]
```diff
@@ -104,6 +105,20 @@ config ZIGBEE_DEBUG_FUNCTIONS
 	  Include functions to suspend/resume zboss thread.
 	  It may be helpful when debugging but using this functions can cause instability of the device.
 
+# Configure default behavior of zb_osif_abort() function called as ZBOSS assert handler
+choice
+	prompt "Default behavior on ZBOSS stack internal assert handler"
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:110]: "on ... handler" sounds weird... "for the ZBOSS stack interna assert handler"?

File: subsys/zigbee/Kconfig [hunk 3]
```diff
@@ -104,6 +105,20 @@ config ZIGBEE_DEBUG_FUNCTIONS
...
+	default ZBOSS_RESET_ON_ASSERT
+
+	config ZBOSS_RESET_ON_ASSERT
+		bool "Reset device when ZBOSS assert occurs"
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:114]: when [a] ZBOSS assert

File: subsys/zigbee/Kconfig [hunk 4]
```diff
@@ -104,6 +105,20 @@ config ZIGBEE_DEBUG_FUNCTIONS
...
+		depends on REBOOT
+
+	config ZBOSS_HALT_ON_ASSERT
+		bool "Halt device when ZBOSS assert occurs, use only for testing and debugging."
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:118]: Halt device when a ZBOSS assert occurs. Use only for testing and debugging.

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by tomchy — APPROVED** [review]
LGTM

**Pull request merged**